### DISCUSSION TITLE
Automate Resampling of Clips for "Simulated Stem Export"

### DISCUSSION THREAD

**Discussion by seangoodvibes** [SynthstromAudible/DelugeFirmware#2254] [Ideas]
### Please describe the problem:

The manual way of exporting clip stems is to resample each clip individually.

### What is the expected behavior?

Add a special playback mode that automates the resampling of clips individually. This special playback mode would not let you do anything else. You would wait until it is done resampling all clips (e.g. display could show # of clips processed / remaining). You could however cancel out of the mode to interrupt the process.

### What hardware did you reproduce it with?

OLED

### What firmware did you reproduce it with?

Built from community

**Comment by sichtbeton** [comment]
Omg. This. Is. So. Good.

**Comment by sichtbeton** [comment]
Complete tracks (for each instrument) would also be sufficient for most I think. Individual clips is nice but it would cut off reverb tails or delays in arrangements.

**Comment by seangoodvibes** [comment]
> Complete tracks (for each instrument) would also be sufficient for most I think. Individual clips is nice but it would cut off reverb tails or delays in arrangements.

Starting with clips and then we can go from there.

**Comment by m-m-adams** [comment]
Converting to discussion - using issues for ideas makes it easy to miss bug reports

**Comment by seangoodvibes** [comment]
Feature PR is ready to go for testing!

https://github.com/SynthstromAudible/DelugeFirmware/pull/2260

**Comment by sichtbeton** [comment]
this is great. a couple of things that could make it even more awesome:

- toggle for normalization OFF, especially for exporting arranger stems (so mixes stay the same when ported to a DAW)
- export dialogue before starting, changes here are temporary for this export only and revert to default on the next one 
- defaults menu for export options 
- silent sidechain triggering, so tracks sound the same when solo-resampled (exported that is), only in arranger
- BPM into file or folder name
- muting all the monitored input: to avoid accidentally recording an input signal onto every exported track (safeguard against this)
- maybe optional audio output mute (as a toggle)

Far future:
- offline rendering faster then realtime...


related feature:
- load a complete stems folder with one action to import all tracks into audio clips (to check the export)
